src/components/selection_input/selection_input.ts

```diff
@@ -116,7 +116,7 @@ export class SelectionInput extends Component<Props, SpreadsheetChildEnv> {
       : this.props.ranges
       ? this.props.ranges.map((xc, id) => ({
           xc,
-          id,
+          id: id + 1,
```

Comment:
@rrahir, We need to begin the ID numbering from 1 in this context. This is because in the SelectionInputPlugin, during the insertNewRange operation, we follow the same convention.

Refer to this: [Link](https://github.com/odoo/o-spreadsheet/blob/saas-16.2/src/plugins/ui_feature/selection_input.ts#L178-L189)

Bcz, when focusing on a range, there won't be any range with ID 0. If this approach seems acceptable to you, we can proceed with the forward port.

Reply:
I think it makes 0 difference (in practise) because we force a rerendering as soon as the component is mounted, which means we'll ahve the proper id on the ranges. I figure the failing test was `Switching from selection input to composer should update the highlihts` but a next tick would have done the trick as well n, which would be closer to the actual behaviour in the client but still - your approach is not bad